single_tech_samples/streamanalytics/README.md

```diff
@@ -1,11 +1,138 @@
 # Azure Stream Analytics
 
-WIP
+![introductory diagram](./docs/images/ASA-job.PNG)
+
+[Azure Stream Analytics](https://azure.microsoft.com/en-us/services/stream-analytics/) is a serverless real-time analytics service. The goal of this sample is to demonstrate how to develop a streaming pipeline, with IaC and testability in mind.
 
 ## Prerequisites
 
 ## Setup
 
+1. __Azure Cli__ Will be necessary for various tasks. Please follow the instructions found [here](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+1. __Bicep__  This project uses `Bicep` templates to setup `Azure` infrastructure. Please follow the steps under [Install and manage via Azure CLI (easiest)](https://github.com/Azure/bicep/blob/main/docs/installing.md#install-and-manage-via-azure-cli-easiest) to install the `Azure Cli` extension.
+
+   A `Bicep` tutorial can be found [here](https://github.com/Azure/bicep/blob/main/docs/tutorial/01-simple-template.md).
+
+1. __Raspberry Pi Azure IoT Online Simulator__ To run through the [Functional Test](#Functional%20Test) section, you can use [this](https://azure-samples.github.io/raspberry-pi-web-simulator/) simulator. Makes sure to follow the step required to edit the js code.
+
+1. __azure-streamanalytics-cicd__ Unit testing is possible using the `npm` package [azure-streamanalytics-cicd](https://www.npmjs.com/package/azure-streamanalytics-cicd). The tool provides additional features for working with `Azure Stream Analytics` locally. Check [here](https://docs.microsoft.com/en-us/azure/stream-analytics/cicd-tools?tabs=visual-studio-code) for more info.
+
+## Build
+
+```bash
+# Run Unit Tests
+azure-streamanalytics-cicd test -project ./asaproj.json -outputPath ./output/
+
+# Create Resource Group
+az group create -n my-rg -l japaneast
+
+# Validate Generated Template
+az deployment group validate -f main.bicep -g my-rg --parameters query='@./streamanalytics-tech-sample.asaql'
+
+# Show plan
+az deployment group what-if -f main.bicep -g my-rg --parameters query='@./streamanalytics-tech-sample.asaql'
+```
+
 ## Running the sample
 
+```bash
+# Create Azure Resources. This will also start the job
+
+az deployment group create -f main.bicep -g my-rg --parameters query='@./streamanalytics-tech-sample.asaql'
+```
+
+## Functional Test
+
+```bash
+# Add device
+
+az iot hub device-identity create --hub-name iot-tech-sample-test --device-id iot-tech-sample-test --edge-enabled
```

Comment:
hub-name will have to line up with whatever the user updated their `param name` with. I think its worth to call out explicitly.

Reply:
Looks like the `az iot` extension is deprecated :|
"This extension 'azure-cli-iot-ext' is deprecated and scheduled for removal. Please remove and add 'azure-iot' instead." --- 
EDIT: aaah, looks like the one I had installed is deprecated, and had to install the azure-iot one (both uses the `az iot` path -- that is confusing :|

Reply:
Good call out here. On further consideration, I think the right thing to do is just remove any confusion and just explicitly set `name` & `env`. And just parameterize all the external commands.